runtime/lua/vim/lsp/util.lua

```diff
@@ -1240,45 +1230,40 @@ local function find_window_by_var(name, value)
   end
 end
 
---- Trims empty lines from input and pad top and bottom with empty lines
----
----@param contents table of lines to trim and pad
----@param opts table with optional fields
----             - pad_top    number of lines to pad contents at top (default 0)
----             - pad_bottom number of lines to pad contents at bottom (default 0)
----@return table table of trimmed and padded lines
-function M._trim(contents, opts)
-  validate({
-    contents = { contents, 't' },
-    opts = { opts, 't', true },
-  })
-  opts = opts or {}
-  contents = M.trim_empty_lines(contents)
-  if opts.pad_top then
-    for _ = 1, opts.pad_top do
-      table.insert(contents, 1, '')
-    end
+---Replaces separator lines by the given divider and removing surrounding blank lines.
+---@param contents string[]
+---@param divider string
+---@return string[]
+local function replace_separators(contents, divider)
+  ---@param line string
+  local function is_blank_line(line)
+    return line and line:match('^%s*$')
   end
-  if opts.pad_bottom then
-    for _ = 1, opts.pad_bottom do
-      table.insert(contents, '')
-    end
+
+  ---@param line string
+  local function is_separator_line(line)
+    return line and line:match('^---+$')
```

Comment:
A bit pedantic, but separators can be made up of `-`, `_`, or `*` (https://github.github.com/gfm/#thematic-breaks). And there can be up to 3 spaces before and any amount of whitespace after.

Also `-` characters have to be escaped (isn't Lua fun? 🙃)

```suggestion
    if line then
      line = line:gsub('^ ? ? ?', ''):gsub('%s*$', '')
      return line:match('^%-%-%-+$') or line:match('^%*%*%*+$') or line:match('^___+$')
    end
    return nil
```

Reply:
Or use `vim.pesc`.

Reply:
Remember, start simple, and extend when someone shows us a server that actually uses an unhandled variant.

Reply:
I don't think this violates "start simple", but at the very least we should handle leading and trailing whitespace:

```lua
return line and line:match('^ ? ? ?%-%-%-+\s*$')
```